test/Microsoft.AspNetCore.Mvc.IntegrationTests/AuthorizeFilterIntegrationTest.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authorization.Infrastructure;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using Microsoft.AspNetCore.Mvc.Authorization;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.IntegrationTests
+{
+    public class AuthorizeFilterIntegrationTest
+    {
+        [Fact]
+        public async Task AuthorizeFilter_CalledTwiceWithNonDefaultProvider()
+        {
+            var applicationModelProviderContext = new ApplicationModelProviderContext(
```

Comment:
Would be nice to see some comments here in the style of `// Arrange, // Act, // Assert` even if you're not rigidly following that flow.